boltdesignsystem/bolt
# ACD-286 Academy popover overflow bug

## Jira

https://pegadigitalit.atlassian.net/browse/ACD-286

## Summary

Add custom Academy component to fix bug where popover gets clipped by `<div>` with overflow "hidden".

## Details

Academy's horizontal card used the grid with `--bordered` class, negative margin, and overflow "hidden" to create a layout with border-separated items. Because this layout requires overflow hidden, any popover inside the content gets clipped. This PR introduces a custom component that uses very simple css grid styles to accomplish the border-separated layout without requiring overflow "hidden".

It also adds a demo of the popover menu in search results: `/pattern-lab/patterns/50-pages-60-academy-05-pages-search-search-results-acd-search-results-page/50-pages-60-academy-05-pages-search-search-results-acd-search-results-page.html`

> Note: due to known z-index issues the popover can never be displayed in the 'bottom' position inside the card. If it is, it will fall behind the next card. I've created this ticket to address the root issue: https://pegadigitalit.atlassian.net/browse/DS-239

## How to test

- Review code changes
- Test demo, see first card in Search results template: `/pattern-lab/patterns/50-pages-60-academy-05-pages-search-search-results-acd-search-results-page/50-pages-60-academy-05-pages-search-search-results-acd-search-results-page.html`. Verify popover menu does not get clipped.
